_test/tests/inc/sort_without_collator.test.php

```diff
@@ -2,161 +2,85 @@
-     * @dataProvider pairs
-     * @param $str1
-     * @param $str2
+     * Since we always use the fallback here, we do not check for
+     * the availability of the intl extension here at all, instead this
+     * test always succeeds
      */
-    public function test_intl_strcmp($str1, $str2) {
-        $this->assertLessThan(0, Sort::strcmp($str1, $str2));
+    public function testIntlExtenstionAvailability()
+    {
+        $this->assertTrue(true);
     }
 
-    /**
-     * @depends test_no_intl_extension
-     */
-    public function test_intl_sort() {
-        $sorted = explode(' ', $this->collation());
-        $random = explode(' ', $this->collation());
-        shuffle($random);
-        Sort::sort($random);
-        $this->assertEquals(array_values($random), array_values($sorted));
-    }
-
-    /**
-     * @depends test_no_intl_extension
-     */
-    public function test_intl_ksort() {
-        $sorted = array_flip(explode(' ', $this->collation()));
-        $random = explode(' ', $this->collation());
-        shuffle($random);
-        $random = array_flip($random);
-        Sort::ksort($random);
-        $this->assertEquals(array_keys($random), array_keys($sorted));
-    }
-
-    /**
-     * @depends test_no_intl_extension
-     */
-    public function test_intl_asort() {
-        $sorted = explode(' ', $this->collation());
-        $keys = array_keys($sorted);
-        shuffle($keys);
-        foreach($keys as $key) $random[$key] = $sorted[$key];
-        Sort::asort($random);
-        $this->assertEquals(array_values($random), array_values($sorted));
-        $this->assertEquals(array_keys($random), array_keys($sorted));
-    }
-
-    /**
-     * @depends test_no_intl_extension
-     */
-    public function test_intl_asortFN_url() {
-        global $conf;
-        $conf['fnencode'] = 'url';
-
-        $sorted = explode('+', urlencode($this->collation()));
-        $keys = array_keys($sorted);
-        shuffle($keys);
-        foreach($keys as $key) $random[$key] = $sorted[$key];
-        Sort::asortFN($random);
-        $this->assertEquals(array_values($random), array_values($sorted));
-        $this->assertEquals(array_keys($random), array_keys($sorted));
-    }
-
-    /**
-     * @depends test_no_intl_extension
-     */
-    public function test_intl_asortFN_safe() {
-        global $conf;
-        $conf['fnencode'] = 'safe';
-
-        $sorted = explode(' ', $this->collation());
-        foreach(array_keys($sorted) as $key) $sorted[$key] = SafeFN::encode($sorted[$key]);
-        $keys = array_keys($sorted);
-        shuffle($keys);
-        foreach($keys as $key) $random[$key] = $sorted[$key];
-        Sort::asortFN($random);
-        $this->assertEquals(array_values($random), array_values($sorted));
-        $this->assertEquals(array_keys($random), array_keys($sorted));
+    /** @inheritDoc */
+    public function provideWordPairs()
+    {
+        // these sorts are mostly wrong. but all our fallback can do
```

Comment:
Please drop this comment, as it does not help anyone to provide examples in other languages.
See below.